### ISSUE TITLE
Import from KeePass

### ISSUE THREAD

**Issue by betsythefc** [beemdevelopment/Aegis#549]
I use KeePassXD to store my TOTP seeds, and it works kind of okay as a 2FA app in itself, but it's user interface is much more devoted to a password manager, than a 2FA app like Aegis. I would like to import from KeePassXD by having Aegis read keepass files, asking for the password, and adding all account with an attribute called `otp` or `TOTP Seed` (The two standards used by KeePassXD). In the `otp` case it's a simple `otpauth://` URI, and all the data is there to setup. with `TOTP Seed`, only the seed is there, but the Issuer, username can be derived from the entry.

**Comment by alexbakker** [comment]
Thanks for the detailed proposal.

This was briefly discussed in #272. We closed that one because we don't want to support saving to KDBX, but that's not what you're requesting. Like I said there, if there's a good existing KDBX library we can use, I would be OK with adding support for importing from KDBX. If not, this would be a lot of work with little reward, in my opinion.

Also curious about @michaelschattgen's thoughts on this.

**Comment by betsythefc** [comment]
Another possible idea, for those of us who are more technically savvy, a converter from kdbx to aegis vault, so we can import the vault file. I'm playing with the idea now.

**Comment by alexbakker** [comment]
@betsythefc Cool! Let us know if you have questions about the Aegis part.

**Comment by NotAFile** [comment]
I personally found importing to be pretty quick via the KeepassXC (desktop) using a search for "attr:totp" and the "show QR code" function (no shortcut sadly). It's not fully automatic of course, but was very doable for the two or so dozen entries I had at the time.

**Comment by NumbGnat** [comment]
> I use KeePassXD to store my TOTP seeds, and it works kind of okay as a 2FA app in itself, but it's user interface is much more devoted to a password manager, than a 2FA app like Aegis. 

I don't know what the best practices are these days, but I can only assume that storing your TOTP in the same app/database together, defeats one of the protections that 2FA provides.

If your password database is compromised, 2FA provides an additional level of protection. If they're stored in the same database, that level of protection is lost. I don't even store my 2FA backups in the same location as my KeePass backups.

**Comment by betsythefc** [comment]
@NumbGnat, I keep them separate, but use it as a database so I have my TOTP codes on my computers/tablets/phones, so they are all synced up. Unless there is a cross-platform TOTP app that can use a common datastore and sync between devices, but I haven't come up with any.

**Comment by alexbakker** [comment]
We currently don't have plans to implement this, so we'll close this.

**Comment by KaKi87** [comment]
Hello @alexbakker,

Why do you not want the app to support importing from KeePass while it does support importing from numerous others, including some that are not even open source ?

Thanks.

**Comment by alexbakker** [comment]
Because supporting the entire spectrum op KDBX versions is a big amount of effort for very little gain.

**Comment by KaKi87** [comment]
How about just the latest ?

**Comment by pglpm** [comment]
I have the same problem here. I use KeePassXC on my Linux laptop, and I'd like to import the tokens from KeePassXC to Aegis on my phone. I tried by exporting from KeePassXC to a simple text file (csv, xml, html), but Aegis doesn't seem to accept it.

This means that at least for the moment I simply have to drop Aegis and find another app that can "collaborate" nicely with KeePassXC. I'd be happy if this possibility were implemented in Aegis.
